.github/workflows/ruff_formater.yml

```diff
@@ -0,0 +1,22 @@
+name: Ruff formater
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
```

Comment:
Do we need to hard code the Python version?

Reply:
Yes, otherwise this can vary across runners and thus change unexpectedly ([docs](https://github.com/actions/setup-python?tab=readme-ov-file#basic-usage)). We can however extend this to a matrix of multiple versions if this is desirable?

Reply:
A matrix would be ideal (HA uses Python 3.12 I think), but this is fine for now

Reply:
Updated pytest with matrix (3.10, 3.11, 3.12) in last commit.